abel/tools/circularize.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+import numpy as np
+from scipy import ndimage
+from scipy.interpolate import UnivariateSpline
+from scipy.optimize import leastsq
+
+import abel
+
+# see https://github.com/PyAbel/PyAbel/issues/186 for discussion
+
+
+def circularize_image(IM, method="lsq", center=None, radial_range=None,
+                      zoom=1, smooth=0, nslices=32, ref_angle=None, 
+                      inverse=False, return_correction=False):
+    """
+    Remove radial distortion from a velocity-map-image through radial scaling
+    of the Newton-rings to enforce circularity.
+
+    The image is divided into angular slices, adjacent radial intensity
+    profile slices compared to give a radial scaling factor that best
+    aligns common structure.
+
+    The resultant radial scaling factor vs angle is interpolated to apply the
+    correction to the whole image grid. The image is remapped to the
+    correct cartesian grid.
+
+    Parameters
+    ----------
+    IM : numpy 2D array
+
+    method: str
+        method to align slice profiles,
+        "argmax" - compare intensity-profile.argmax() of each radial slice.
```

Comment:
I think that the least-squares method is likely to produce good results in a broad range of situations. However, the argmax method relies on the image having an obvious sharp peak, right? I think that we should make this clear:

"a very-fast but very-naive alignment method that aligns the adjacent slices based on the most intense pixel in each slice. This method may provide good results for images that have one sharp, intense radial peak. Otherwise it will not produce correct results." 

I also think that there would be a similar algorithm that would be essentially a combination of both methods:

calculate the least-squares error for a range of scaling factors (say, `linspace(0.9,1.1,20)`)
now, find the argmax of this array. Or, better yet, up-sample a bit and find the best scaling factor from a spline fit. 
Basically, this a nearly identical approach to what we are doing for `lsq`, except that we can try a bunch of scaling factors in parallel. It's probably much faster than `lsq`, and unlikely to catastophically fail like `least_squares` can sometimes do. But, it will be less precise.